dubbo-remoting/dubbo-remoting-http12/src/main/java/org/apache/dubbo/remoting/http12/netty4/h1/NettyHttp1Codec.java

```diff
@@ -80,19 +86,28 @@ private void doWriteHeader(ChannelHandlerContext ctx, HttpMetadata msg, ChannelP
         if (CollectionUtils.isNotEmpty(statusHeaders)) {
             status = HttpResponseStatus.valueOf(Integer.parseInt(statusHeaders.get(0)));
         }
+        if (keepAlive) {
+            headers.add(HttpHeaderNames.CONNECTION.getKey(), String.valueOf(HttpHeaderValues.KEEP_ALIVE));
+        } else {
+            headers.add(HttpHeaderNames.CONNECTION.getKey(), String.valueOf(HttpHeaderValues.CLOSE));
+        }
         // process normal headers
-        ctx.writeAndFlush(new DefaultHttpResponse(HttpVersion.HTTP_1_1, status, headers.getHeaders()), promise);
+        ctx.write(new DefaultHttpResponse(HttpVersion.HTTP_1_1, status, headers.getHeaders()), promise);
     }
 
     private void doWriteMessage(ChannelHandlerContext ctx, HttpOutputMessage msg, ChannelPromise promise) {
         if (HttpOutputMessage.EMPTY_MESSAGE == msg) {
-            ctx.writeAndFlush(LastHttpContent.EMPTY_LAST_CONTENT, promise);
+            if (!keepAlive) {
```

Comment:
Not flipping the boolean will make the code better understood:
···
            if (keepAlive) {
                ctx.writeAndFlush(LastHttpContent.EMPTY_LAST_CONTENT, promise);
            } else {
                ctx.writeAndFlush(LastHttpContent.EMPTY_LAST_CONTENT, promise).addListener(ChannelFutureListener.CLOSE);
            }
···

Reply:
> Not flipping the boolean will make the code better understood:
> 
> ```java
>             if (keepAlive) {
>                 ctx.writeAndFlush(LastHttpContent.EMPTY_LAST_CONTENT, promise);
>             } else {
>                 ctx.writeAndFlush(LastHttpContent.EMPTY_LAST_CONTENT, promise).addListener(ChannelFutureListener.CLOSE);
>             }
> ```

done